association_backend/association/domain/entities/subscription_entities.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+import pydantic
+from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Table
+from sqlalchemy.orm import registry, relationship
+
+
+class UserData(pydantic.BaseModel):
+    email: str
+    user_id: int
+
+
+class SubscriptionState(str, Enum):
+    PENDING = "pending"
+    ACTIVE = "active"
+    EXPIRED = "expired"
+    NOT_CREATED = "not-created"
+
+    def __str__(self) -> str:
+        return str.__str__(self)
+
+
+@dataclass
+class Subscription:
+    user_id: int
+    creation_date: datetime
+    state: SubscriptionState
+    # is_for_life: bool
+    user_email: str = ""
```

Comment:
why is this optional? and why we need it?

Reply:
I deleted it